modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1581,23 +1581,64 @@ private ObjectEntry _toObjectEntry(
 					serviceBuilderObjectEntry, dtoConverterContext.getUriInfo())
 			).build();
 
+			boolean siteScoped = Objects.equals(
+				objectDefinition.getScope(),
+				ObjectDefinitionConstants.SCOPE_SITE);
+
+			HashMap<String, String> templateParameterMap = HashMapBuilder.put(
+				() -> {
+					if (siteScoped) {
+						return "scopeKey";
+					}
+
+					return null;
+				},
+				() -> {
+					if (siteScoped) {
+						return String.valueOf(
+							serviceBuilderObjectEntry.getGroupId());
+					}
+
+					return null;
+				}
+			).put(
+				() -> {
+					if (siteScoped) {
+						return "externalReferenceCode";
+					}
+
+					return "objectEntryExternalReferenceCode";
+				},
+				serviceBuilderObjectEntry.getExternalReferenceCode()
+			).build();
+
 			for (ObjectAction objectAction :
 					_objectActionLocalService.getObjectActions(
 						objectDefinition.getObjectDefinitionId(),
 						ObjectActionTriggerConstants.KEY_STANDALONE)) {
 
+				String methodName = null;
+
+				if (siteScoped) {
```

Comment:
This could be calculated also out of the loop, otherwise you would be executing N times the same condition